packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -108,10 +108,21 @@ abstract interface class RenderAbstractViewport extends RenderObject {
   /// when the offset of the viewport is changed by x then `target` also moves
   /// by x within the viewport.
   ///
+  /// The optional [Axis] is used by
+  /// [RenderTwoDimensionalViewport.getOffsetToReveal] to
```

Comment:
Some where in this doc should mention that it can't be null when calling RenderTwoDimensionalViewport.getOffsetToReveal.

This is a bit weird though. The caller may not know which subclass the object is. Either all the subclass can handle null case or we force the type to be non null.

Reply:
This seems no addressed yet

Reply:
Oh! I rewrote the docs. I has to be nullable or it's a breaking change to require it, there was another comment thread here, it may have gotten lost..

Reply:
I did't want to require folks to provide the axis, it does not make sense to do so when it's 1D, there's only one Axis. I updated the docs and added assertions based on your other comment.